tests/reg_issue3191/test_preconditions_of_numerical_conversion.fz

```diff
@@ -0,0 +1,36 @@
+# This file is part of the Fuzion language implementation.
+#
+# The Fuzion language implementation is free software: you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as published
+# by the Free Software Foundation, version 3 of the License.
+#
+# The Fuzion language implementation is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU General Public License along with The
+# Fuzion language implementation.  If not, see <https://www.gnu.org/licenses/>.
+
+
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion test
+#
+# -----------------------------------------------------------------------
+
+reg_issue3191 is
+
+  a i32 := 256
+  b u8 := a.as_u8       # should produce an error due to a failed precondition
+  say "$a.as_u8 is $b"
+
+  c i64 := -1
+  d u32 := c.as_u32     # same here
```

Comment:
I think it is better to copy the comment above.
First it is easier to understand, second if the example is extended then the comment might become wrong.